src/backend/distributed/commands/utility_hook.c

```diff
@@ -468,6 +468,18 @@ ProcessUtilityInternal(PlannedStmt *pstmt,
 		PreprocessTruncateStatement((TruncateStmt *) parsetree);
 	}
 
+	if (IsA(parsetree, LockStmt))
+	{
+		LockStmt *stmt = (LockStmt *) parsetree;
+
+		ereport(NOTICE, errmsg("Processing LOCK command."));
+
+		ErrorIfUnsupportedLockStmt(stmt);
+		uint32 nowaitFlag = stmt->nowait ? DIST_LOCK_NOWAIT : 0;
+		AcquireDistributedLockOnRelations(stmt->relations, stmt->mode,
+										  DIST_LOCK_VIEWS_RECUR | nowaitFlag);
```

Comment:
do we need to filter out non-Citus tables?

Reply:
They are filtered inside `AcquireDistributedLockOnRelations`.
`AcquireDistributedLockOnRelations` now does a first layer processing of relations based on the config flags that it takes.
Then after it extracts the relations that need to be locked, it passes them to `AcquireDistributedLockOnRelations_Internal` to do pretty much what the old `AcquireDistributedLockOnRelations` was doing.